doc/manual/source/conf.py

```diff
@@ -7,13 +7,13 @@
 # -- Path setup --------------------------------------------------------------
 
 # If extensions (or modules to document with autodoc) are in another directory,
-# add these directories to sys.path here. If the directory is relative to the
+# add these directories to pathlib.Path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
 import sys
 import os
-sys.path.insert(0, os.path.abspath('../..'))
-sys.path.insert(0, os.path.abspath('../../src'))
+pathlib.Path.insert(0, os.path.abspath('../..'))
```

Comment:
this isn't how you do it.  Please try and figure out how to do it differently using pathlib.

check that the docs build locally before pushing so we don't use CI credits for debugging but only testing debugged code.  Also, there is still an os.path in there .....

Reply:
maybe ask chatgpt what that old syntax is doing so you understand that, then read a bit about pathlib so you know hte background and then trh and figure out how to do it with pathlib (and in the end ask chatgpt how how it would do it).....   (giving away my secrets here...don't let chatgpt take over, but get help)

Reply:
Remember to also refactor these `../..` into `../../..`

Reply:
so we will keep `sys.path`, there is no pathlib replacement for that, it contains the list of directories the system looks in in to find executables, but `os.path.abspath` will go to `str(Path('../..').resolve())`  (but get the right number of parent directories so everything works. and I think we need the `str()`)

Reply:
You still require `sys.path` as that gives the interpreter knowledge of where to look for modules. The update is to remove uses of `os`. Though generally fine to use, `os` handles paths as strings, while `Pathlib` creates a platform-specific object with some additional functionalities. Probably the most useful is `glob` if you have worked with the `*` operator on bash.